### PR TITLE
Move from Swift to Obj-C

### PR DESCRIPTION
**Pull request by long1eu**
https://github.com/kiwi-bop/flutter_crashlytics/issues/43

### CODE REVIEW

**Comment by jaumard** [comment]
Nice !!! Thanks ! I'll try to review and test it asap ! Keep you updated !

**Comment by jaumard** [comment]
Hey @long1eu I've retrieve your fork in order to test if it works or not, and I can't make the example compile anymore :/ do you manage to run it on your side ?

**Comment by jaumard** [comment]
apparently even if I add them back, it's still complaining about a missing file `main.m`. Let me know when you have the example working @long1eu thanks ! From what I can see the code is fine, now need to check if it works

File: example/ios/Podfile.lock [hunk 1]
```diff
@@ -27,8 +27,8 @@ SPEC CHECKSUMS:
...
   Fabric: a2917d3895e4c1569b9c3170de7320ea1b1e6661
   Flutter: 9d0fac939486c9aba2809b7982dfdbb47a7b0296
-  flutter_crashlytics: 753059d75232a519279a515b51e01e7f784e5385
+  flutter_crashlytics: 5db12dd8bfe2ce59bb579894ba79e054f3df7666
 
-PODFILE CHECKSUM: ac8aca132925cdff4493e5b19d72732202854236
+PODFILE CHECKSUM: aff02bfeed411c636180d6812254b2daeea14d09
 
-COCOAPODS: 1.5.3
+COCOAPODS: 1.5.2
```
[the remark this answers is not in the document]
Comment (long1eu) [reply example/ios/Podfile.lock]: Sure. I'll make it happen.

**Comment by long1eu** [comment]
> apparently even if I add them back, it's still complaining about a missing file `main.m`. Let me know when you have the example working @long1eu thanks ! From what I can see the code is fine, now need to check if it works

I seems like those files were not committed.

**Comment by jaumard** [comment]
Example is running but crash reporting is not, it's `trace` not `traces` but even by fixing this it's still crashing after @long1eu

**Comment by jaumard** [comment]
Almost all good @long1eu , but it still crash when you report a crash manually. 
```
 Runner[94181:470244] -[NSNull UTF8String]: unrecognized selector sent to instance 0x11177af28
Runner[94181:470244] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSNull UTF8String]: unrecognized selector sent to instance 0x11177af28'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001115081bb __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x0000000110aae735 objc_exception_throw + 48
	2   CoreFoundation                      0x0000000111526f44 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   CoreFoundation                      0x000000011150ced6 ___forwarding___ + 1446
	4   CoreFoundation                      0x000000011150eda8 _CF_forwarding_prep_0 + 120
	5   Runner                              0x000000010e1bc77a _Z17CLSExceptionWriteP7CLSFile16CLSExceptionTypePKcS3_P7NSArray + 760
	6   Runner                              0x000000010e1c3f48 __CLSUserLoggingWriteAndCheckABFiles_block_invoke + 57
	7   libdispatch.dylib                   0x000000010efd6602 _dispatch_client_callout + 8
	8   libdispatch.dylib                   0x000000010efe4653 _dispatch_lane_barrier_sync_invoke_and_complete + 132
	9   Runner                              0x000000010e1c36e7 CLSUserLoggingWriteAndCheckABFiles + 164
	10  Runner                              0x000000010e1bc410 CLSExceptionRecord + 342
	11  Runner                              0x000000010e1a7272 -[Crashlytics recordCustomExceptionName:reason:frameArray:] + 228
	12  Runner                              0x000000010e1a30fc -[FlutterCrashlyticsPlugin onInitialisedMethodCall:result:] + 1084
	13  Runner                              0x000000010e1a2c5d -[FlutterCrashlyticsPlugin handleMethodCall:result:] + 429
	14  Flutter                             0x000000010f2e89a2 __45-[FlutterMethodChannel setMethodCallHandler:]_block_invoke + 115
	15  Flutter                             0x000000010f305616 _ZNK5shell21PlatformMessageRouter21HandlePlatformMessageEN3fml6RefPtrIN5blink15PlatformMessageEEE + 166
	16  Flutter                             0x000000010f308fbe _ZN5shell15PlatformViewIOS21HandlePlatformMessageEN3fml6RefPtrIN5blink15PlatformMessageEEE + 38
	17  Flutter                             0x000000010f3598e9 _ZNSt3__110__function6__funcIZN5shell5Shell29OnEngineHandlePlatformMessageEN3fml6RefPtrIN5blink15PlatformMessageEEEE4$_26NS_9allocatorIS9_EEFvvEEclEv + 57
	18  Flutter                             0x000000010f3144fa _ZN3fml15MessageLoopImpl15RunExpiredTasksEv + 922
	19  Flutter                             0x000000010f317832 _ZN3fml17MessageLoopDarwin11OnTimerFireEP16__CFRunLoopTimerPS0_ + 26
	20  CoreFoundation                      0x000000011146df34 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20
	21  CoreFoundation                      0x000000011146db32 __CFRunLoopDoTimer + 1026
	22  CoreFoundation                      0x000000011146d39a __CFRunLoopDoTimers + 266
	23  CoreFoundation                      0x0000000111467a1c __CFRunLoopRun + 2252
	24  CoreFoundation                      0x0000000111466e11 CFRunLoopRunSpecific + 625
	25  GraphicsServices                    0x0000000114c7d1dd GSEventRunModal + 62
	26  UIKitCore                           0x0000000119def81d UIApplicationMain + 140
	27  Runner                              0x000000010e1a2560 main + 112
	28  libdyld.dylib                       0x000000010f04c575 start + 1
	29  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

**Comment by long1eu** [comment]
I was just looking at this. Thanks.

**Comment by long1eu** [comment]
according to [this](https://flutter.io/docs/development/platform-integration/platform-channels) null is nil and NSNull in objc. So I need to check for NSNull in nested objects (like maps) and nil otherwise.

**Comment by jaumard** [comment]
That's strange that I didn't had the same issue in swift...

**Comment by long1eu** [comment]
It's not that strange since AFAIK swift does this kind of stuff for you.

**Pull request merged**